### ISSUE TITLE
Add auto completion system

### ISSUE THREAD

**Issue by SadE54** [shibayan/Sharprompt#197]
I will use the library for a specific domain language CLI, but it could be cool and to a system for autocompletion  ! 
Anyway thx for the lib :)

**Comment by genyx** [comment]
This would be cool for a file-browser functionality

**Comment by ProphetLamb** [comment]
On that note, for the free text `Input` a history would be quite nice, so that we can navigate prev/next using arrow keys. Ill see if I can hack something together

**Comment by Mart-Bogdan** [comment]
Yeah, would be nice to provide callback delegate for user code to provide completions and list of options for completions.
Both ways would have it's uses.
